typhon/cloudmask/aster.py

```diff
@@ -100,12 +100,26 @@ def read_digitalnumbers(self, channel):
             ndarray: Digital numbers. Shape according to data resolution of
                 the respective channel.
         """
-        if channel in ('1', '2', '3N', '3B'):
-            subsensor = 'VNIR'
+        if channel in ('1', '2', '3N'):
+            if self.get_metadata()['ASTEROBSERVATIONMODE.1']=='VNIR1, ON':
```

Comment:
You could use ``self.meta`` as it is defined in when initialising the ``ASTERimage``.